contracts/test/RevenueDistributionToken.t.sol

```diff
@@ -24,30 +24,82 @@ contract ConstructorTest is TestUtils {
 
 }
 
-contract DepositAndMintWithPermitTest is TestUtils {
+contract RDTTestBase is TestUtils {
+
+    /***********************/
+    /*** Setup Variables ***/
+    /***********************/
 
     MockERC20 asset;
     RDT       rdToken;
 
-    uint256 stakerPrivateKey    = 1;
-    uint256 notStakerPrivateKey = 2;
-    uint256 nonce               = 0;
-    uint256 deadline            = 5_000_000_000;  // Timestamp far in the future
+    uint256 nonce    = 0;
+    uint256 deadline = 5_000_000_000;  // Timestamp far in the future
 
     uint256 constant sampleAssetsToConvert = 1e18;
     uint256 constant sampleSharesToConvert = 1e18;
 
-    address staker;
-    address notStaker;
+    bytes constant ARITHMETIC_ERROR = abi.encodeWithSignature("Panic(uint256)", 0x11);
+
+    uint256 constant START = 10_000_000;
 
     function setUp() public virtual {
         asset   = new MockERC20("MockToken", "MT", 18);
         rdToken = new RDT("Revenue Distribution Token", "RDT", address(this), address(asset), 1e30);
 
+        vm.warp(START);  // Warp to non-zero timestamp
+    }
+
+    // Deposit asset into RDT
+    function _depositAsset(address asset_, address staker_, uint256 depositAmount_) internal {
+        MockERC20(asset_).mint(staker_, depositAmount_);
+        Staker(staker_).erc20_approve(asset_, address(rdToken), depositAmount_);
+        Staker(staker_).rdToken_deposit(address(rdToken), depositAmount_);
+    }
+
+    // Transfer funds into RDT and update the vesting schedule
+    function _transferAndUpdateVesting(address asset_, address rdToken_, uint256 vestingAmount_, uint256 vestingPeriod_) internal {
```

Comment:
I think you can remove `asset` and `rdToken` from the input parameters since they're already state variables.
Same for all the other utility functions where state variables are available.